### ISSUE TITLE
Outdated build instructions

### ISSUE THREAD

**Issue by wlnirvana** [sharkdp/cube-composer#44]
Since this project relies on some outdated versions of packages (like purescript@0.11.6, purescript-arrays@4.1.2, etc) , the [local install instruction][1] no longer works now.

I've managed to compile the project by manually locking the package to the old versions in `package.json` and `bower.json`, and am wondering if a PR is acceptable for this approach.

[1]:https://github.com/sharkdp/cube-composer#local-install

**Comment by sharkdp** [comment]
That would be great, yes. Thank you.

**Comment by tomyuval** [comment]
The version of PureScript is not pinned in `package.json`, and the project doesn't build with the current version (0.14.5). Looking for a solution I found this open issue, and based on the comment above I thought (part of) the solution would be to pin PureScript to 0.11.6. However, `npm` fails to install that version.

The earliest PureScript version `npm` did manage to install for me is `0.12.0` – but that already breaks the build.

I'm new to this repository and also to PureScript; I intend to dig a little into it and see if I can work something out, but maybe someone else who's more familiar (e.g. @sharkdp, @wlnirvana) can lend a hand in moving this forward?

Let's get this repository working again!

**Comment by tomyuval** [comment]
So, in order to use PureScript 0.12.0, I updated the dependencies to the following:

```
    "purescript-arrays": "^5.0.0",
    "purescript-console": "^4.0.0",
    "purescript-enums": "^4.0.0",
    "purescript-foldable-traversable": "^4.0.0",
    "purescript-integers": "^4.0.0",
    "purescript-math": "^2.0.0",
    "purescript-nullable": "^4.0.0",
    "purescript-strings": "^4.0.0",
```

Note I removed the deprecated `purescript-dom`, which [only works with versions prior to 0.12.x](https://github.com/purescript-deprecated/purescript-dom).

I also had a problem with `purescript-isometric`, whose latest release (3.0.0) is older than 0.12.x, and whose inclusion caused _a lot_ of version conflicts in dependencies. But the [GitHub repository](https://github.com/sharkdp/purescript-isometric/) has a newer state which just hasn't been released (yet), so I solved the issue temporarily by taking `src` – along with the dependencies in `bower.json` – from the [current snapshot](https://github.com/sharkdp/purescript-isometric/tree/f026cb7f9ba7ec969e758fc1facf4f774128319e) and embedding them in (my local copy of) this code base.

With that I managed to `bower install`.

But many things in the code base need to be updated in order for it to compile:
* anything using the deprecated [`purescript-eff`](https://github.com/purescript-deprecated/purescript-eff) (`Control.Monad.Eff`);
* anything using the deprecated [`purescript-maps`](https://github.com/purescript-deprecated/purescript-maps) (`Data.StrMap`);
* anything using the deprecated [`purescript-dom`](https://github.com/purescript-deprecated/purescript-dom);
* possibly other stuff.

Being unfamiliar with the language and with this code base, this is a bit too much for me to tackle on my own. @sharkdp, would you like to pair up on it? Hopefully, after an initial joint session to introduce me to some of the specifics I should be able to continue by myself.

Or maybe someone else can do it.

Or alternatively, maybe someone can find a way to install PureScript 0.11.x – this should make everything much simpler.

**Comment by sharkdp** [comment]
Thank you for looking into this, @tomyuval. I'm not very active in the PureScript community anymore and therefore not familiar with recent updates to all the core libraries and the tooling changes. I currently don't have the time to help with getting this up & running again, but I should find some time to review PRs, if you (or someone else) wants to help.

Maybe you could ask the PureScript community for help? They have been very supportive in the past. Not sure where the best place to ask would be though. Maybe https://www.reddit.com/r/purescript/?

**Comment by tomyuval** [comment]
Thanks, @sharkdp.
Eventually I figured it out: pinning PureScript to 0.11.6 _does_ work (what happened before is I used `^0.11.6` which was resolved to 0.11.7, and _that_ didn't work).
With #45 I managed to build the project locally without any problem.
